File: cinema-backend/src/main/java/org/example/cinemabackend/cinema/application/dto/request/CreateCinemaRequest.java
package org.example.cinemabackend.cinema.application.dto.request;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import org.example.cinemabackend.user.application.dto.response.CinemaManagerResponse;

import java.util.Set;

@Builder
public record CreateCinemaRequest(
        @NotBlank String name,
        @NotBlank String description,
        @NotNull CreateAddressRequest address,
        @NotNull CreateImageRequest image,
        @NotNull Set<@NotNull CreateScreeningRoomRequest> screeningRooms,
        @NotNull Set<@NotNull CreateContactDetailsRequest> contactDetails,
        @NotNull @Valid CinemaManagerResponse cinemaManager
) {
}

File: cinema-frontend/src/app/cinema/dtos/request/create-cinema.request.ts
import {CreateScreeningRoomRequest} from "./create-screening-room.request";
import {CreateContactDetailsRequest} from "./create-contact-details.request";
import {CreateImageRequest} from "../../../movie/dtos/request/create-image.request";
import {CreateAddressRequest} from "./create-address.request";
import {UserResponse} from "../../../user/dtos/response/user.response";

export class CreateCinemaRequest {

  constructor(
    readonly name: string,
    readonly description: string,
    readonly address: CreateAddressRequest,
    readonly image: CreateImageRequest,
    readonly screeningRooms: CreateScreeningRoomRequest[],
    readonly contactDetails: CreateContactDetailsRequest[],
    readonly cinemaManager: UserResponse
  ) {

  }

}
